sunpy/coordinates/frames.py

```diff
@@ -125,6 +125,14 @@ def represent_as(self, base, s='base', in_frame_units=False):
             data.lon.wrap_angle = self._wrap_angle
         return data
 
+    @property
+    def size(self):
+        """
+        Returns the size of the underlying data if it exists, else returns 0.  This overrides the
+        property in `~astropy.coordinates.BaseCoordinateFrame`.
+        """
+        return self.data.size if self.has_data else 0
```

Comment:
Why this change?

Reply:
For any frame with data, `astropy.coordinates` checks that any frame attributes that are non-scalar (i.e., size > 1) are able to be broadcast to the shape of the data.  See https://github.com/astropy/astropy/blob/03dfb7b5005f13bb975e6d6c5b0d60ca9e44a866/astropy/coordinates/baseframe.py#L614.

If a frame attribute is itself a frame, its size is checked to see if it's non-scalar.  However, the `BaseCoordinateFrame.size` property doesn't check whether the data exists before querying the data's size (see https://github.com/astropy/astropy/blob/03dfb7b5005f13bb975e6d6c5b0d60ca9e44a866/astropy/coordinates/baseframe.py#L707).  That means that a data-less frame raises an error when used as an attribute for a frame with data.

In this implementation/API, I specifically need to be able to precisely that.  That's why I modified `SunPyBaseCoordinateFrame.size` to return zero if there is no data.  As long as code doesn't query a frame's size until after it confirms that there is frame data, there's no difference in behavior.  It's such a subtle issue that I've haven't yet bothered opening an issue upstream to discuss it.